### PR TITLE
zos: fix setsockopt error when using AF_UNIX

### PR DESCRIPTION
**Pull request by miladfarca**

### CODE REVIEW

File: src/unix/udp.c [hunk 1]
```diff
@@ -263,11 +263,29 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
...
  * on other platforms so we don't enable it.
+ *
+ * zOS does not support getsockname with SO_REUSEPORT option when using 
+ * AF_UNIX  
  */
 static int uv__set_reuse(int fd) {
   int yes;
 
-#if defined(SO_REUSEPORT) && !defined(__linux__)
+#if defined(__MVS__) && defined(SO_REUSEPORT)
```
Comment (sam-github) [inline src/unix/udp.c]: nit: maybe reorder this so SO_REUSEPORT is first, as it is in the other conditional? Just for consistency, the compiler won't care.

File: src/unix/udp.c [hunk 2]
```diff
@@ -263,11 +263,29 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
...
+ *
+ * zOS does not support getsockname with SO_REUSEPORT option when using 
+ * AF_UNIX  
  */
 static int uv__set_reuse(int fd) {
   int yes;
 
-#if defined(SO_REUSEPORT) && !defined(__linux__)
+#if defined(__MVS__) && defined(SO_REUSEPORT)                  
+  yes = 1;
```
Comment (sam-github) [inline src/unix/udp.c]: pico-nit: I don't see why `yes` wasn't initialized to `1` at the time of its declaration in the original code

**Comment by mhdawson** [comment]
Launched https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1274/

**Comment by mhdawson** [comment]
One more try to launch CI https://ci.nodejs.org/job/libuv-test-commit-zos/1441/

File: src/unix/udp.c [hunk 3]
```diff
@@ -263,11 +263,29 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
...
+  struct sockaddr_in sockfd;
+  unsigned int sockfd_len = sizeof (sockfd);
+  if (getsockname(fd, (struct sockaddr *) &sockfd, &sockfd_len) == -1) {
+      return UV__ERR(errno);
+  }
```
Comment (santigimeno) [inline src/unix/udp.c]: style: remove brackets in one-line conditionals

File: src/unix/udp.c [hunk 4]
```diff
@@ -263,11 +263,29 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
...
+  if (sockfd.sin_family == AF_UNIX){
+    if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes)))
```
Comment (santigimeno) [inline src/unix/udp.c]: nit: space before the curly bracket

File: src/unix/udp.c [hunk 5]
```diff
@@ -263,11 +263,29 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
...
+      return UV__ERR(errno);
+  }
+  else {
```
Comment (santigimeno) [inline src/unix/udp.c]: style: `} else {`

**Comment by miladfarca** [comment]
Styling issues have been addressed, initiating `yes` to `1` in the beginning of the function

File: src/unix/udp.c [hunk 6]
```diff
@@ -263,16 +263,29 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
  * are different from the BSDs: it _shares_ the port rather than steal it
  * from the current listener.  While useful, it's not something we can emulate
  * on other platforms so we don't enable it.
+ *
+ * zOS does not support getsockname with SO_REUSEPORT option when using 
+ * AF_UNIX  
  */
 static int uv__set_reuse(int fd) {
-  int yes;
+  int yes = 1;
```
Comment (cjihrig) [inline src/unix/udp.c]: Can you keep the variable declaration and assignment separate.

File: src/unix/udp.c [hunk 7]
```diff
@@ -263,16 +263,29 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
...
 
-#if defined(SO_REUSEPORT) && !defined(__linux__)
-  yes = 1;
+#if defined(SO_REUSEPORT) && defined(__MVS__)                 
+  struct sockaddr_in sockfd;
+  unsigned int sockfd_len = sizeof (sockfd);
+  if (getsockname(fd, (struct sockaddr *) &sockfd, &sockfd_len) == -1)
```
Comment (cjihrig) [inline src/unix/udp.c]: No space before the star in the cast please.

File: src/unix/udp.c [hunk 8]
```diff
@@ -263,16 +263,30 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
  * are different from the BSDs: it _shares_ the port rather than steal it
  * from the current listener.  While useful, it's not something we can emulate
  * on other platforms so we don't enable it.
+ *
+ * zOS does not support getsockname with SO_REUSEPORT option when using 
+ * AF_UNIX
```
Comment (saghul) [inline src/unix/udp.c]: nit: punctuation.

Comment (miladfarca) [reply src/unix/udp.c]: '.' has been added to the end of the sentence.

**Comment by mhdawson** [comment]
@cjihrig does it look good to you know?

File: src/unix/udp.c [hunk 9]
```diff
@@ -263,16 +263,30 @@ static void uv__udp_sendmsg(uv_udp_t* handle) {
...
  */
 static int uv__set_reuse(int fd) {
   int yes;
-
-#if defined(SO_REUSEPORT) && !defined(__linux__)
   yes = 1;
+
+#if defined(SO_REUSEPORT) && defined(__MVS__)                 
+  struct sockaddr_in sockfd;
+  unsigned int sockfd_len = sizeof (sockfd);
```
Comment (cjihrig) [inline src/unix/udp.c]: No space after `sizeof`.

**Comment by cjihrig** [comment]
@mhdawson yes.

File: src/unix/udp.c (hunk 9 above)
Comment (miladfarca) [reply src/unix/udp.c]: space has been removed.

**Comment by cjihrig** [comment]
CI with no related failures: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1353/

Landed in https://github.com/libuv/libuv/commit/fd67cd538cf9cde9fa79530bb969f26fddaec3ee. Thanks!

**Pull request closed without merging**